multiqc/modules/sourmash/gather.py

```diff
@@ -0,0 +1,221 @@
+        percents as values. These values are used to identify the top 5 matches
+        identified across all samples.
+        """
+        for s_name, data in self.gather_raw_data.items():
+            for row in data:
+                match_name = row["match_name"]
+                # loop over all samples and sum different variables
+                if match_name not in self.gather_pct_per_match_all_samples:
+                    self.gather_pct_per_match_all_samples[match_name] = 0
+                self.gather_pct_per_match_all_samples[match_name] += row["pct_unique_weighted"]
+
+    def calculate_pct_unclassified_per_sample(self):
+        """
+        Calculate the percent of each sample that was unclassified.
+        """
+        for s_name, data in self.gather_raw_data.items():
+            for row in data:
+                # sum over all matches for a given query to calculate the percent classified
+                if s_name not in self.gather_pct_unclassified_per_sample:
+                    self.gather_pct_unclassified_per_sample[s_name] = 0
+                self.gather_pct_unclassified_per_sample[s_name] += row["pct_unique_weighted"]
+            # convert to % unclassified
+            self.gather_pct_unclassified_per_sample[s_name] = 100 - self.gather_pct_unclassified_per_sample[s_name]
+
+    def calculate_pct_top_five_per_sample(self):
+        """
+        Calculate the percent of each sample that is attributable to the top 5 genomes
+        across all samples. The output is a dictionary with samples as keys and
+        the summed percent for the top 5 genomes as values.
+        """
+        # get top genomes matched across samples
+        sorted_pct = sorted(self.gather_pct_per_match_all_samples.items(), key=lambda x: x[1], reverse=True)
+        for pct_sum in sorted_pct[: self.top_n]:
+            self.gather_top_five_matches.append(pct_sum[0])  # append the genome name to the top five list
+
+        # calculate the pct attributable to the top 5 matches per sample
+        for match_name in self.gather_top_five_matches:
+            for s_name, data in self.gather_raw_data.items():
+                if s_name not in self.gather_pct_top_five_per_sample:
+                    self.gather_pct_top_five_per_sample[s_name] = 0
+                for row in data:
+                    if row["match_name"] == match_name:
+                        self.gather_pct_top_five_per_sample[s_name] += row["pct_unique_weighted"]
+
+    def general_stats_columns(self):
+        """
+        add columns to the general statistics table for % top 5 matches and % unclassified
+        """
+        # Column headers
+        headers = OrderedDict()
+        headers["% Top 5"] = {
+            "title": "% Top 5 Genomes",
+            "description": "Percentage of sample that was classified as one of the top 5 genomes ({})".format(
+                ", ".join(self.gather_top_five_matches)
+            ),
```

Comment:
Should probably be an f-string, but still..
```suggestion
            "description": "Percentage of sample that was classified as one of the top {} genomes ({})".format(
                self.top_n,
                ", ".join(self.gather_top_five_matches)
            ),
```